ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -839,8 +859,11 @@ func isPIInstanceRefreshFunc(client *st.IBMPIInstanceClient, id, instanceReadySt
 		if err != nil {
 			return nil, "", err
 		}
+		if (*pvm.Status == "SHUTOFF") && (pvm.Health.Status == instanceReadyStatus || pvm.Health.Status == helpers.PIInstanceHealthOk) {
```

Comment:
```suggestion
		if (*pvm.Status == "SHUTOFF") {
```
You don't need to do any health status checks here, right?

Reply:
You still do, because they can go error state for example.

Reply:
There's some code in line 1019 that looks like
```
		if *pvm.Status == "SHUTOFF" && pvm.Health.Status == helpers.PIInstanceHealthOk {
			log.Printf("The lpar is now off after the resource change...")
			return pvm, "SHUTOFF", nil
		}
```
I'm assuming we should be emulating that?